sdk/src/logs/logger.cc

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/sdk/logs/logger.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+Logger::Logger() noexcept : processor_{nullptr} {}
```

Comment:
Are there any use cases for this constructor?
Since a Logger is always created by` LoggerProvider::GetLogger()`, which always calls the `Logger(processor)` constructor, should this constructor be removed, along with `Logger::GetProcessor()` and `Logger::SetProcessor()`?

Reply:
I don't see a need for the default constructor. We should always initialize a logger and a logger provider with a processor.

The `SetProcessor` method should stay though, as one might possibly want to change the processor.